diff --git a/arch/arm64/boot/dts/apple/t8132.dtsi b/arch/arm64/boot/dts/apple/t8132.dtsi
index bbe47e5..962adb5 100644
--- a/arch/arm64/boot/dts/apple/t8132.dtsi
+++ b/arch/arm64/boot/dts/apple/t8132.dtsi
@@ -8,6 +8,7 @@
 #include <dt-bindings/interrupt-controller/apple-aic.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/pinctrl/apple.h>
+#include <dt-bindings/phy/phy.h>
 
 / {
 	compatible = "apple,t8132", "apple,arm-platform";
@@ -492,6 +493,110 @@ serial0: serial@3ad200000 {
 			status = "disabled";
 		};
 
+		dwc3_0: usb@402280000 {
+			compatible = "apple,t8132-dwc3", "apple,t8103-dwc3";
+			reg = <0x4 0x02280000 0x0 0xcd00>,
+			      <0x4 0x0228cd00 0x0 0x3200>;
+			reg-names = "dwc3-core", "dwc3-apple";
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1279 IRQ_TYPE_LEVEL_HIGH>;
+			iommus = <&dwc3_0_dart_0 0>,
+			         <&dwc3_0_dart_1 1>;
+			power-domains = <&ps_atc0_usb>;
+			resets = <&atcphy0>;
+			phys = <&atcphy0 PHY_TYPE_USB2>, <&atcphy0 PHY_TYPE_USB3>;
+			phy-names = "usb2-phy", "usb3-phy";
+			status = "disabled";
+		};
+
+		dwc3_0_dart_0: iommu@402f00000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x4 0x02f00000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1283 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&ps_atc0_usb>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		dwc3_0_dart_1: iommu@402f80000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x4 0x02f80000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1283 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&ps_atc0_usb>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		atcphy0: phy@403000000 {
+			compatible = "apple,t8132-atcphy", "apple,t8122-atcphy";
+			reg = <0x4 0x03000000 0x0 0x4c000>,
+			      <0x4 0x03050000 0x0 0x8000>,
+			      <0x4 0x00000000 0x0 0x8000>,
+			      <0x4 0x02a90000 0x0 0x4000>,
+			      <0x4 0x02a84000 0x0 0x4000>;
+			reg-names = "core", "lpdptx", "axi2af", "usb2phy",
+			            "pipehandler";
+
+			#phy-cells = <1>;
+			#reset-cells = <0>;
+
+			power-domains = <&ps_atc0_usb>;
+		};
+
+		dwc3_1: usb@40a280000 {
+			compatible = "apple,t8132-dwc3", "apple,t8103-dwc3";
+			reg = <0x4 0x0a280000 0x0 0xcd00>,
+			      <0x4 0x0a28cd00 0x0 0x3200>;
+			reg-names = "dwc3-core", "dwc3-apple";
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1341 IRQ_TYPE_LEVEL_HIGH>;
+			iommus = <&dwc3_1_dart_0 0>,
+			         <&dwc3_1_dart_1 1>;
+			power-domains = <&ps_atc1_usb>;
+			resets = <&atcphy1>;
+			phys = <&atcphy1 PHY_TYPE_USB2>, <&atcphy1 PHY_TYPE_USB3>;
+			phy-names = "usb2-phy", "usb3-phy";
+			status = "disabled";
+		};
+
+		dwc3_1_dart_0: iommu@40af00000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x4 0x0af00000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1345 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&ps_atc1_usb>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		dwc3_1_dart_1: iommu@40af80000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x4 0x0af80000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1345 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&ps_atc1_usb>;
+			#iommu-cells = <1>;
+			status = "disabled";
+		};
+
+		atcphy1: phy@40b000000 {
+			compatible = "apple,t8132-atcphy", "apple,t8122-atcphy";
+			reg = <0x4 0x0b000000 0x0 0x4c000>,
+			      <0x4 0x0b050000 0x0 0x8000>,
+			      <0x4 0x08000000 0x0 0x8000>,
+			      <0x4 0x0aa90000 0x0 0x4000>,
+			      <0x4 0x0aa84000 0x0 0x4000>;
+			reg-names = "core", "lpdptx", "axi2af", "usb2phy",
+			            "pipehandler";
+
+			#phy-cells = <1>;
+			#reset-cells = <0>;
+
+			power-domains = <&ps_atc1_usb>;
+		};
+
 		ans_mbox: mbox@481608000 {
 			compatible = "apple,t8132-asc-mailbox", "apple,asc-mailbox-v4";
 			reg = <0x4 0x81608000 0x0 0x4000>;
