diff --git a/arch/arm64/boot/dts/apple/t8132.dtsi b/arch/arm64/boot/dts/apple/t8132.dtsi
index 70257f7d..8934bc1e 100644
--- a/arch/arm64/boot/dts/apple/t8132.dtsi
+++ b/arch/arm64/boot/dts/apple/t8132.dtsi
@@ -5,6 +5,7 @@
  * Copyright The Asahi Linux Contributors
  */
 
+#include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/interrupt-controller/apple-aic.h>
 #include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/pinctrl/apple.h>
@@ -430,6 +431,10 @@ i2c4_pins: i2c4-pins {
 				pinmux = <APPLE_PINMUX(108, 1)>,
 				         <APPLE_PINMUX(107, 1)>;
 			};
+
+			pcie_pins: pcie-pins {
+				pinmux = <APPLE_PINMUX(160, 2)>;
+			};
 		};
 
 		sep_dart: iommu@3a8c80000 {
@@ -815,6 +820,113 @@ nvme: nvme@485cc0000 {
 			resets = <&ps_ans>;
 			status = "disabled";
 		};
+
+		pcie0_dart: iommu@490000000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x4 0x90000000 0x0 0x4000>;
+			#iommu-cells = <1>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1232 IRQ_TYPE_LEVEL_HIGH>;
+			power-domains = <&ps_apcie_phy_sw>;
+			status = "disabled";
+		};
+
+		pcie0: pcie@1cb0000000 {
+			compatible = "apple,t8132-pcie", "apple,t6020-pcie";
+			device_type = "pci";
+
+			reg = <0x1c 0xb0000000 0x0 0x10000000>,
+			      <0x4 0x94000000 0x0 0x4000>,
+			      <0x4 0x90028000 0x0 0x8000>,
+			      <0x4 0x91028000 0x0 0x8000>,
+			      <0x4 0x92028000 0x0 0x8000>,
+			      <0x4 0x97020000 0x0 0x4000>,
+			      <0x4 0x97024000 0x0 0x4000>,
+			      <0x4 0x97028000 0x0 0x4000>;
+			reg-names = "config", "rc", "port0", "port1", "port2",
+				"phy0", "phy1", "phy2";
+
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1231 IRQ_TYPE_LEVEL_HIGH>,
+				<AIC_IRQ 1240 IRQ_TYPE_LEVEL_HIGH>,
+				<AIC_IRQ 1249 IRQ_TYPE_LEVEL_HIGH>;
+
+			msi-controller;
+			msi-parent = <&pcie0>;
+			msi-ranges = <&aic AIC_IRQ 1506 IRQ_TYPE_EDGE_RISING 96>;
+
+			iommu-map = <0x100 &pcie0_dart 1 1>,
+				<0x101 &pcie0_dart 2 1>;
+			iommu-map-mask = <0xffff>;
+
+			bus-range = <0 8>;
+			#address-cells = <3>;
+			#size-cells = <2>;
+			ranges = <0x43000000 0xb 0xc0000000 0xb 0xc0000000 0x0 0x20000000>,
+				 <0x02000000 0x0 0x80000000 0xb 0x80000000 0x0 0x40000000>;
+
+			power-domains = <&ps_apcie_phy_sw>;
+			pinctrl-0 = <&pcie_pins>;
+			pinctrl-names = "default";
+			status = "disabled";
+
+			port00: pci@0,0 {
+				device_type = "pci";
+				reg = <0x0 0x0 0x0 0x0 0x0>;
+				reset-gpios = <&pinctrl_ap 163 GPIO_ACTIVE_LOW>;
+
+				#address-cells = <3>;
+				#size-cells = <2>;
+				ranges;
+
+				interrupt-controller;
+				#interrupt-cells = <1>;
+
+				interrupt-map-mask = <0 0 0 7>;
+				interrupt-map = <0 0 0 1 &port00 0 0 0 0>,
+					<0 0 0 2 &port00 0 0 0 1>,
+					<0 0 0 3 &port00 0 0 0 2>,
+					<0 0 0 4 &port00 0 0 0 3>;
+			};
+
+			port01: pci@1,0 {
+				device_type = "pci";
+				reg = <0x800 0x0 0x0 0x0 0x0>;
+
+				#address-cells = <3>;
+				#size-cells = <2>;
+				ranges;
+
+				interrupt-controller;
+				#interrupt-cells = <1>;
+
+				interrupt-map-mask = <0 0 0 7>;
+				interrupt-map = <0 0 0 1 &port01 0 0 0 0>,
+					<0 0 0 2 &port01 0 0 0 1>,
+					<0 0 0 3 &port01 0 0 0 2>,
+					<0 0 0 4 &port01 0 0 0 3>;
+				status = "disabled";
+			};
+
+			port02: pci@2,0 {
+				device_type = "pci";
+				reg = <0x1000 0x0 0x0 0x0 0x0>;
+
+				#address-cells = <3>;
+				#size-cells = <2>;
+				ranges;
+
+				interrupt-controller;
+				#interrupt-cells = <1>;
+
+				interrupt-map-mask = <0 0 0 7>;
+				interrupt-map = <0 0 0 1 &port02 0 0 0 0>,
+					<0 0 0 2 &port02 0 0 0 1>,
+					<0 0 0 3 &port02 0 0 0 2>,
+					<0 0 0 4 &port02 0 0 0 3>;
+				status = "disabled";
+			};
+		};
 	};
 };
 
