diff --git a/arch/arm64/boot/dts/apple/t8132-j713.dts b/arch/arm64/boot/dts/apple/t8132-j713.dts
index 5975c2d11..58440190f 100644
--- a/arch/arm64/boot/dts/apple/t8132-j713.dts
+++ b/arch/arm64/boot/dts/apple/t8132-j713.dts
@@ -18,6 +18,10 @@ / {
 	compatible = "apple,j713", "apple,t8132", "apple,arm-platform";
 	model = "Apple MacBook Air (13-inch, M4, 2025)";
 
+	aliases {
+		keyboard = &keyboard;
+	};
+
 	led-controller {
 		compatible = "pwm-leds";
 		led-0 {
@@ -79,4 +83,43 @@ &nvme {
 	status = "okay";
 };
 
+&mtp {
+	status = "okay";
+};
+
+&mtp_mbox {
+	status = "okay";
+};
+
+&mtp_dart {
+	status = "okay";
+};
+
+&mtp_dockchannel {
+	status = "okay";
+};
+
+&mtp_hid {
+	apple,afe-reset-gpios = <&smc_gpio 8 GPIO_ACTIVE_LOW>;
+	apple,stm-reset-gpios = <&smc_gpio 24 GPIO_ACTIVE_LOW>;
+
+	multi-touch {
+		firmware-name = "apple/tpmtfw-j713.bin";
+	};
+
+	keyboard: keyboard {
+		hid-country-code = <0>;
+		apple,keyboard-layout-id = <0>;
+	};
+
+	stm {
+	};
+
+	actuator {
+	};
+
+	tp_accel {
+	};
+};
+
 #include "hwmon-laptop.dtsi"
diff --git a/arch/arm64/boot/dts/apple/t8132.dtsi b/arch/arm64/boot/dts/apple/t8132.dtsi
index 8934bc1ea..3389bc910 100644
--- a/arch/arm64/boot/dts/apple/t8132.dtsi
+++ b/arch/arm64/boot/dts/apple/t8132.dtsi
@@ -386,6 +386,80 @@ pinctrl_aop: pinctrl@390824000 {
 			             <AIC_IRQ 429 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
+		mtp: mtp@394600000 {
+			compatible = "apple,t8132-mtp", "apple,t8132-rtk-helper-asc4",
+				     "apple,mtp", "apple,rtk-helper-asc4";
+			reg = <0x3 0x94600000 0x0 0x4000>,
+			      <0x3 0x94050000 0x0 0x4000>;
+			reg-names = "asc", "sram";
+			mboxes = <&mtp_mbox>;
+			iommus = <&mtp_dart 0>;
+			#helper-cells = <0>;
+			status = "disabled";
+		};
+
+		mtp_mbox: mbox@394608000 {
+			compatible = "apple,t8132-asc-mailbox", "apple,asc-mailbox-v4";
+			reg = <0x3 0x94608000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1114 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 1113 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 1116 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 1115 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "send-empty", "send-not-empty",
+					  "recv-empty", "recv-not-empty";
+			#mbox-cells = <0>;
+			status = "disabled";
+		};
+
+		mtp_dart: iommu@394800000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x3 0x94800000 0x0 0x4000>;
+			#iommu-cells = <1>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1091 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
+		};
+
+		mtp_dockchannel: fifo@394b14000 {
+			compatible = "apple,t8132-dockchannel", "apple,dockchannel";
+			reg = <0x3 0x94b14000 0x0 0x1000>;
+			reg-names = "irq";
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 1093 IRQ_TYPE_LEVEL_HIGH>;
+
+			ranges = <0x0000 0x3 0x94b28000 0x1000>,
+				 <0x4000 0x3 0x94b2c000 0x1000>,
+				 <0x8000 0x3 0x94b30000 0x1000>,
+				 <0xc000 0x3 0x94b34000 0x1000>;
+			nonposted-mmio;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			interrupt-controller;
+			#interrupt-cells = <2>;
+
+			status = "disabled";
+
+			mtp_hid: input@8000 {
+				compatible = "apple,dockchannel-hid";
+				reg = <0x8000 0x1000>,
+				      <0xc000 0x1000>,
+				      <0x0000 0x1000>,
+				      <0x4000 0x1000>;
+				reg-names = "config", "data",
+					    "rmt-config", "rmt-data";
+				iommus = <&mtp_dart 0>;
+				interrupt-parent = <&mtp_dockchannel>;
+				interrupts = <2 IRQ_TYPE_LEVEL_HIGH>,
+					     <3 IRQ_TYPE_LEVEL_HIGH>;
+				interrupt-names = "tx", "rx";
+
+				apple,fifo-size = <0x800>;
+				apple,helper-cpu = <&mtp>;
+			};
+		};
+
 		pinctrl_ap: pinctrl@39a000000 {
 			compatible = "apple,t8132-pinctrl", "apple,t8103-pinctrl";
 			reg = <0x3 0x9a000000 0x0 0x100000>;
