diff --git a/arch/arm64/boot/dts/apple/t8132.dtsi b/arch/arm64/boot/dts/apple/t8132.dtsi
index e5636c4d7..a7e08ef43 100644
--- a/arch/arm64/boot/dts/apple/t8132.dtsi
+++ b/arch/arm64/boot/dts/apple/t8132.dtsi
@@ -365,6 +365,64 @@ pinctrl_smc: pinctrl@38c820000 {
 			             <AIC_IRQ 558 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
+		aop_mbox: mbox@390608000 {
+			compatible = "apple,t8132-asc-mailbox", "apple,asc-mailbox-v4";
+			reg = <0x3 0x90608000 0x0 0x4000>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 434 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 433 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 436 IRQ_TYPE_LEVEL_HIGH>,
+				     <AIC_IRQ 435 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "send-empty", "send-not-empty",
+					  "recv-empty", "recv-not-empty";
+			#mbox-cells = <0>;
+			status = "disabled";
+		};
+
+		aop_dart: iommu@390f00000 {
+			compatible = "apple,t8132-dart", "apple,t8110-dart";
+			reg = <0x3 0x90f00000 0x0 0x4000>;
+			#iommu-cells = <1>;
+			interrupt-parent = <&aic>;
+			interrupts = <AIC_IRQ 457 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
+		};
+
+		aop_admac: dma-controller@393300000 {
+			compatible = "apple,t8132-admac", "apple,t8122-admac";
+			reg = <0x3 0x93300000 0x0 0x34000>;
+			#dma-cells = <1>;
+			dma-channels = <9>;
+			interrupts-extended = <0>,
+					      <0>,
+					      <&aic AIC_IRQ 478 IRQ_TYPE_LEVEL_HIGH>,
+					      <0>;
+			iommus = <&aop_dart 10>;
+			status = "disabled";
+		};
+
+		aop: aop@390c00000 {
+			compatible = "apple,t8132-aop", "apple,t6020-aop";
+			reg = <0x3 0x90c00000 0x0 0x1e0000>,
+			      <0x3 0x90600000 0x0 0x4000>;
+			mboxes = <&aop_mbox>;
+			mbox-names = "mbox";
+			iommus = <&aop_dart 0>;
+
+			status = "disabled";
+
+			aop_audio: audio {
+				compatible = "apple,t8132-aop-audio", "apple,t6030-aop-audio";
+				dmas = <&aop_admac 1>;
+				dma-names = "dma";
+			};
+
+			aop_als: als {
+				compatible = "apple,t8132-aop-als", "apple,aop-als";
+				// intentionally empty
+			};
+		};
+
 		pinctrl_aop: pinctrl@390824000 {
 			compatible = "apple,t8132-pinctrl", "apple,t8103-pinctrl";
 			reg = <0x3 0x90824000 0x0 0x4000>;
